File: Day 2/DayTwo.js
function productExceptSelf(nums) {
    const n = nums.length;
    if (n === 0) return [];

    const leftProducts = new Array(n).fill(1);
    const rightProducts = new Array(n).fill(1);
    const result = new Array(n).fill(1);

    // Fill the leftProducts array
    for (let i = 1; i < n; i++) {
        leftProducts[i] = leftProducts[i - 1] * nums[i - 1];
    }

    // Fill the rightProducts array
    for (let i = n - 2; i >= 0; i--) {
        rightProducts[i] = rightProducts[i + 1] * nums[i + 1];
    }

    // Fill the result array by multiplying left and right products
    for (let i = 0; i < n; i++) {
        result[i] = leftProducts[i] * rightProducts[i];
    }

    return result;
}

console.log(productExceptSelf([1, 2, 3, 4, 5])); 
console.log(productExceptSelf([3, 2, 1]));

File: Day 2/DayTwo.py
def product_except_self(nums):
    n = len(nums)
    if n == 0:
        return []
    
    # Initialize the left and right product arrays
    left_products = [1] * n
    right_products = [1] * n
    result = [1] * n
    
    # Fill the left_products array
    for i in range(1, n):
        left_products[i] = left_products[i - 1] * nums[i - 1]
    
    # Fill the right_products array
    for i in range(n - 2, -1, -1):
        right_products[i] = right_products[i + 1] * nums[i + 1]
    
    # Fill the result array by multiplying left and right products
    for i in range(n):
        result[i] = left_products[i] * right_products[i]
    
    return result

print(product_except_self([1, 2, 3, 4, 5]))
print(product_except_self([3, 2, 1]))
